xsl/xccdf-resources/openscap.js

```diff
@@ -108,12 +108,131 @@ function ruleSearch()
         $("#search-matches").html("No rules match your search criteria!");
 }
 
+var is_original = true;
+var original_treetable = null;
+
 $(document).ready( function() {
     $("#result-details").hide();
     $(".js-only").show();
     $(".toggle-rule-display").each(function(){
         toggleRuleDisplay(this);
     });
-
-    $(".treetable").treetable({ column: 0, expandable: true, initialState : "expanded", indent : 0 });
+    original_treetable = $(".treetable").clone();
+    $(".treetable").treetable({ column: 0, expandable: true, clickableNodeNames: true, initialState : "expanded", indent : 0 });
+    is_original = true;
 });
+
+function Reset() {
+	if (!is_original) {
+		$(".treetable").remove();
+		$("#rule-overview").append(original_treetable.clone());
+		$(".treetable").treetable({ column: 0, expandable: true, clickableNodeNames: true, initialState : "expanded", indent : 0 });
+		$(".toggle-rule-display").each(function(){
+			toggleRuleDisplay(this);
+		});
+		is_original = true;
+	}
+}
+
+function NewGroupLine(group_name)
+{
+	return "<tr class=\"rule-overview-inner-node\" data-tt-id=\"" + group_name + "\">" +
+		"<td colspan=\"3\"><strong>" + group_name + "</strong></td></tr>";
+}
+
+var KeysEnum = {
+	DEFAULT: "default",
+	SEVERITY : "severity",
+	RESULT: "result",
+	NIST: "data-nist-id",
+	DISA: "data-disa-id"
+};
+
+function GetTargetGroupsList(rule, key)
+{
+	/* This function returns an array of target groups indentifiers */
+	if (key == KeysEnum.SEVERITY) {
+		var severity = rule.children("td:nth-child(2)").text();
+		return [severity];
+	} else if (key == KeysEnum.RESULT) {
+		var result = rule.children(".rule-result").text();
+		return [result];
+	}
+	else {
```

Comment:
IMHO "} else {" would looks better, but replacing "if else" with "switch" would be the best way.